src/sentry/incidents/endpoints/organization_alert_rule_details.py

```diff
@@ -138,6 +138,18 @@ class OrganizationAlertRuleDetailsEndpoint(OrganizationAlertRuleEndpoint):
         "PUT": ApiPublishStatus.UNKNOWN,
     }
 
+    def check_project_access(func):
+        def wrapper(self, request: Request, organization, alert_rule):
+            project = alert_rule.snuba_query.subscriptions.get().project
```

Comment:
why can't you just use `alert_rule.project`? Also, should this endpoint in general just be scoped to a project which will do the permissions checks automatically?

Reply:
lol I _wish_ there was a `project` row in the `alertrule` (metric alerts) table. The model was written so that it can be scoped to all projects in the org, but never actually implemented that way - it's a pain.

A little while back I soft deprecated (I couldn't fully delete them because the Terraform plugin uses some of the endpoints) the project scoped rule and alert rule endpoints because they were essentially duplicated by the organization ones - they did the same thing with some tiny differences and it meant we had to maintain the same code in 2 places and was just generally confusing. The one thing I unfortunately missed was that a user who is not a member of a given project can do crud methods in a project they don't belong to - @oioki brought it up with me.

Reply:
got it. thanks for that clarification. `alert_rule` = metrics `rule` = non-metric alert. was looking at the wrong model. 👍

Reply:
Yeah the naming is horrible 🙃

Reply:
it may be worth adding a comment about how in practice an alert_rule is only ever associated w/ one project at the moment.